lib/elastomer/version_support.rb

```diff
@@ -0,0 +1,48 @@
+module Elastomer
+  class VersionSupport
+    attr_reader :client
+
+    def initialize(client)
+      @client = client
+    end
```

Comment:
My only thought at the moment is this ... could we pass in the version String here instead of a Client instance, and them move all version related checks into this class. This includes the `es_version_2_x?` and `es_version_5_x?` methods.

This change will make this class easier to test since it no longer depends upon a Client connected to a real Elasticsearch cluster. Passing in a String is much easier to stub out.

Reply:
Yeah, good idea. Will update it.